worlds/spelunker/__init__.py

```diff
@@ -0,0 +1,149 @@
+from .Options import SpelunkerOptions
+from .Client import SpelunkerClient
+from .Rules import set_rules
+from .Rom import LocalRom, patch_rom, get_base_rom_path, SpelunkerProcPatch, USHASH
+
+
+class SpelunkerSettings(settings.Group):
+    class RomFile(settings.UserFilePath):
+        """File name of the Spelunker NES ROM"""
+        description = "Spelunker ROM File"
+        copy_to = "Spelunker.nes"
+        md5s = [USHASH]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+
+
+class SpelunkerWeb(WebWorld):
+    theme = "dirt"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Spelunker randomizer and connecting to an Archipelago server.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Pink Switch"]
+    )
+
+    tutorials = [setup_en]
+
+
+class SpelunkerWorld(World):
+    """
+    Spelunker is a 2D platformer for the NES.
+    Explore the cavernous depths, collecting treasure along the way and seeking the mythical Pyramid of Gold.
+    Also, you'll probably die a lot.
+    """
+    game = "Spelunker"
+    option_definitions = SpelunkerOptions
+    required_client_version = (0, 4, 6)
+
+    item_name_to_id = {item: item_table[item].code for item in item_table}
+    location_name_to_id = {location.name: location.code for location in get_locations(None)}
+    item_name_groups = get_item_names_per_category()
+
+    web = SpelunkerWeb()
+    settings: typing.ClassVar[SpelunkerSettings]
+    # topology_present = True
+
+    options_dataclass = SpelunkerOptions
+    options: SpelunkerOptions
+
+    locked_locations: List[str]
+    rom_name: bytearray
+
+    def __init__(self, multiworld: MultiWorld, player: int):
+        self.rom_name_available_event = threading.Event()
+        super().__init__(multiworld, player)
+        self.locked_locations = []
+
+    def create_item(self, name: str) -> Item:
+        data = item_table[name]
+        return Item(name, data.classification, data.code, self.player)
+
+    def create_regions(self) -> None:
+        init_areas(self, get_locations(self))
+
+    def get_filler_item_name(self) -> str:
+        if self.random.randint(0, 100) <= 10:
+            return self.random.choice(useful_items)
+        else:
+            return self.random.choice(filler_items)
+
+    def set_rules(self) -> None:
+        set_rules(self)
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("Golden Pyramid", self.player)
+        self.get_location("Golden Pyramid").place_locked_item(self.create_item("Golden Pyramid"))
+
+    def get_excluded_items(self) -> Set[str]:
+        excluded_items: Set[str] = set()
+        return excluded_items
+
+    def create_item_with_correct_settings(self, name: str) -> Item:
+        data = item_table[name]
+        item = Item(name, data.classification, data.code, self.player)
+
+        if not item.advancement:
+            return item
+
+        return item

```

Comment:
```suggestion
    def create_item_with_correct_settings(self, name: str) -> Item:
        data = item_table[name]
        item = Item(name, data.classification, data.code, self.player)
        return item

```

The if statement isn't relevant, since it just returns item both ways

Reply:
Fixed